text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+If multiple conditions are specified in a given retry strategy,
+they must all match for the action to be taken; the conditions are ANDed together, not ORed.
+This example configures four retry strategies:
+
+```ts
+const jobDefn = new batch.EcsJobDefinition(this, 'JobDefn', {
+   containerDefinition: new batch.ContainerDefinition(this, 'containerDefn', {
+    image: ecs.ContainerImage.fromRegistry('public.ecr.aws/amazonlinux/amazonlinux:latest'),
+    memoryLimitMiB: 2048,
+    compatibility: Compatibility.EC2, // specify FARGATE for Fargate workflows
+  }),
+  attempts: 5,
+  retryStrategies: [{
+    action: Action.EXIT,
+    onExitCode: '40*', // matches 40, 400, 404, 40230498, etc
+  }],
+});
+
+jobDefn.addRetryStrategy({
+  action: Action.RETRY,
+  onStatusReason: StatusReason.SPOT_INSTANCE_RECLAIMED,
+});
+
+jobDefn.addRetryStrategy({
+   action: Action.EXIT,
+   onReason: Reason.CANNOT_PULL_CONTAINER,
+});
+
+jobDefn.addRetryStrategy({
+  action: Action.RETRY,
+  onStatusReason: StatusReason.custom('Some other reason*'),
+});
+```
+
+### Running single-container ECS workflows
+
+Batch supports ECS workflows natively. This examples creates a `JobDefinition` that runs a single container with ECS:
+
+```ts
+const jobDefn = new batch.EcsJobDefinition(this, 'JobDefn', {
+  containerDefinition: new batch.ContainerDefinition(this, 'containerDefn', {
+    image: ecs.ContainerImage.fromRegistry('public.ecr.aws/amazonlinux/amazonlinux:latest'),
+    memoryLimitMiB: 2048,
+    compatibility: Compatibility.EC2, // specify FARGATE for Fargate workflows
+  }),
+});
+```
+
+For workflows that need persistent storage, batch supports mounting `Volume`s to the container.
+You can both provision the volume and mount it to the container in a single operation:
+
+```ts
+jobDefn.addVolume({
+  name: 'myVolume',
+  efsVolumeConfiguration: {
+    fileSystem: myFileSystem
+  },
+  containerPath: '/Volumes/myVolume',
+});
+```
+
+### Running Kubernetes Workflows:
+
+Batch also supports running workflows on EKS. The following example creates a `JobDefinition` that runs on EKS:
+
+```ts
+const eksContainer = new batch.EksContainerDefinition(this, 'myEksContainer', {
+   image: ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+
+const eksPod = new batch.EksPod(this, 'myEksPod', {
+   containers: [eksContainer],
+});
+
+new batch.EksJobDefinition(this, 'myEksJobDefn', {
+   pod: eksPod,
+});
+
+// alternative using convienience methods:
+const pod = new batch.EksPod(this, 'myEksPod');
+const jobDefn = new batch.EksJobDefinition(this, 'myEksJobDefn', {
+  pod,
+});
+
+jobDefn.addContainer('myEksContainer', { // adds to the Pod
+  image: ecs.ContainerImage.fromRegistry('my-registry/my-image:latest'),
+});
+```
+
+You can mount `Volume`s to these containers in a single operation:
+
+```ts
+jobDefn.addEmptyDirVolume({
+  name: 'emptyDir',
+  mountPath: '/Volumes/emptyDir',
+});
+jobDefn.addHostPathVolume({
+  name: 'hostPath',
+  hostPath: '/sys', 
```

Comment:
```suggestion
  volume: Volume.hostPath('/sys'),
```

Reply:
Not sure I understand this one, it would become:

```
jobDefn.addHostPathVolume({
  name: 'hostPath',
  volume: Volume.hostPath('/sys'),
  mountPath: '/Volumes/hostPath',
});
```
I could see:

```
Volume.hostPath({
  name: 'hostPath',
  hostPath: '/sys',
  ...
});